scripts/generate_learning_path_markdown.js

```diff
@@ -8,16 +8,48 @@ const getContributors = require('./get_contributors')
 const mkdirSync = require('./mkdir_sync')
 const getArticleFiles = require('./get_article_files')
 const writeMarkdownFile = require('./write_markdown_file')
-
 const sections = require('./section_data.json')
 
 const urls = YAML.parse(fs.readFileSync(join('..', 'config', 'urls.yaml'), 'utf-8'))
 
 const getYouTubeCode = (section, articleNumber) => {
-  const firstEntryOfGroupIndex = urls.findIndex(entry => entry.section === section.toLowerCase())
-  const currentPageIndexOffset = articleNumber - 1
-  const youtubeUrl = urls[firstEntryOfGroupIndex + currentPageIndexOffset].video.youtube
-  return youtubeUrl.replace('https://www.youtube.com/watch?v=', '')
+  const sectionLinks = urls.filter(entry => entry.section === section.toLowerCase())
+  const videoUrls = sectionLinks[articleNumber - 1]
+  if (videoUrls && videoUrls.video && videoUrls.video.youtube) {
+    return videoUrls.video.youtube.replace('https://www.youtube.com/watch?v=', '')
+  }
+  return ''
+}
+
+const getYouTubeImage = (youTubeCode) => {
+  if (youTubeCode) {
+    return `https://img.youtube.com/vi/${youTubeCode}/mqdefault.jpg`
+  }
+  return ''
+}
+
+const getArticleTitle = (article) => {
+  const asciiTitle = article.asciiDoc.match(/== (.*)/)
+  if (asciiTitle) {
+    return asciiTitle[1]
+  }
+  const mdTitle = article.asciiDoc.match(/# (.*)/)
+  if (mdTitle) {
+    return mdTitle[1]
+  }
+  return "TITLE MISSING"
+}
+
+const getArticleText = (article) => {
+  const asciiTitle = article.asciiDoc.match(/== (.*)/)
+  if (asciiTitle) {
+    return article.asciiDoc.replace(/== (.*)/, '')
+  }
+  const mdTitle = article.asciiDoc.match(/# (.*)/)
```

Comment:
Can remove this MD stuff after https://github.com/InnerSourceCommons/InnerSourceLearningPath/pull/505 is merged.